Tools/autotest/arducopter.py

```diff
@@ -4684,14 +4708,25 @@ def loiter_requires_position(self):
         self.context_push()
         self.set_parameter("GPS_TYPE", 0)
         self.reboot_sitl()
-        self.change_mode('STABILIZE')
-        self.wait_ekf_flags((mavutil.mavlink.ESTIMATOR_ATTITUDE |
+
+        # set EKF3 sources to not use GPS in order to avoid pre-arm check failures
+        ahrs_ekf_type = self.get_parameter("AHRS_EKF_TYPE")
+        if ahrs_ekf_type == 3:
+            self.set_parameter("EK3_SRC1_POSXY", 0)
+            self.set_parameter("EK3_SRC1_VELXY", 0)
+            self.set_parameter("EK3_SRC1_VELZ", 0)
+
+        # check for expected EKF flags
+        expected_ekf_flags = (mavutil.mavlink.ESTIMATOR_ATTITUDE |
                              mavutil.mavlink.ESTIMATOR_VELOCITY_VERT |
                              mavutil.mavlink.ESTIMATOR_POS_VERT_ABS |
-                             mavutil.mavlink.ESTIMATOR_CONST_POS_MODE |
-                             mavutil.mavlink.ESTIMATOR_PRED_POS_HORIZ_REL),
-                            0,
-                            timeout=120)
+                             mavutil.mavlink.ESTIMATOR_CONST_POS_MODE)
+        if ahrs_ekf_type == 2:
+            expected_ekf_flags = expected_ekf_flags | mavutil.mavlink.ESTIMATOR_PRED_POS_HORIZ_REL
```

Comment:
worth asking @priseborough about this for EKF2

Reply:
I've raised issue https://github.com/ArduPilot/ardupilot/issues/16018 and will discuss with PaulR